moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp

```diff
@@ -136,21 +138,29 @@ void planning_scene_monitor::CurrentStateMonitor::clearUpdateCallbacks()
 
 void planning_scene_monitor::CurrentStateMonitor::startStateMonitor(const std::string& joint_states_topic)
 {
+  rclcpp::Clock ros_clock;
   if (!state_monitor_started_ && robot_model_)
   {
     joint_time_.clear();
     if (joint_states_topic.empty())
-      ROS_ERROR("The joint states topic cannot be an empty string");
+    {
+      RCLCPP_ERROR(node_->get_logger(), " The joint states topic cannot be an empty string");
+    }
     else
-      joint_state_subscriber_ = nh_.subscribe(joint_states_topic, 25, &CurrentStateMonitor::jointStateCallback, this);
+      joint_state_subscriber_ = node_->create_subscription<sensor_msgs::msg::JointState>(
+          joint_states_topic,
+          std::bind(&planning_scene_monitor::CurrentStateMonitor::jointStateCallback, this, std::placeholders::_1),
+          rmw_qos_profile_sensor_data);
     if (tf_buffer_ && !robot_model_->getMultiDOFJointModels().empty())
     {
-      tf_connection_.reset(new TFConnection(
-          tf_buffer_->_addTransformsChangedListener(boost::bind(&CurrentStateMonitor::tfCallback, this))));
+      // TODO (anasarrak): replace this for the appropiate function, there is no similar
```

Comment:
There should be one now

Reply:
They replaced it [see](https://github.com/ros-planning/moveit/pull/830#issuecomment-386517416), I should change MoveIt to use the new API and will port it to MoveIt2